src/test/java/api/requests/scenarios/LoanDueDatesAfterRecallTests.java

```diff
@@ -847,12 +845,59 @@ public void shouldNotExtendLoanDueDateIfOverdueLoanIsRecalled() {
     ));
   }
 
+  @Test
+  public void shouldExtendLoanDueDateIfOverdueLoanIsRecalledAndConditionsAreMet() {
+    final IndividualResource smallAngryPlanet = itemsFixture.basedUponSmallAngryPlanet();
+
+    final Period alternateLoanPeriod = Period.weeks(3);
+
+    final Period loanPeriod = Period.weeks(1);
+    setFallbackPolicies(new LoanPolicyBuilder()
+      .withName("Can Circulate Rolling With Recalls")
+      .withDescription("Can circulate item With Recalls")
+      .rolling(loanPeriod)
+      .unlimitedRenewals()
+      .renewFromSystemDate()
+      .withAllowsAlternateRecallReturnInterval(true)
+      .withAlternateRecallReturnInterval(alternateLoanPeriod));
+
+    final DateTime loanCreateDate = now(UTC)
+      .minus(loanPeriod.timePeriod())
+      .minusMinutes(1);
+    DateTime expectedLoanDueDate = loanCreateDate
+      .plus(loanPeriod.timePeriod())
+      .plus(alternateLoanPeriod.timePeriod())
+      .plusMinutes(1);
+
+    final IndividualResource loan = checkOutFixture.checkOutByBarcode(
+      smallAngryPlanet, usersFixture.steve(), loanCreateDate);
+
+    requestsFixture.place(new RequestBuilder()
+      .recall()
+      .forItem(smallAngryPlanet)
+      .fulfilToHoldShelf()
+      .by(usersFixture.jessica())
+      .fulfilToHoldShelf()
+      .withPickupServicePointId(servicePointsFixture.cd1().getId()));
+
+    DateTime actualLoanDueDate = DateTime.parse(loansStorageClient.getById(loan.getId()).getJson().getString("dueDate"));
+
+    actualLoanDueDate = actualLoanDueDate
+      .minusSeconds(actualLoanDueDate.getSecondOfMinute())
+      .minusMillis(actualLoanDueDate.getMillisOfSecond());
+
+    expectedLoanDueDate = expectedLoanDueDate
+      .minusSeconds(expectedLoanDueDate.getSecondOfMinute())
+      .minusMillis(expectedLoanDueDate.getMillisOfSecond());
+
+    assertThat(actualLoanDueDate, is(expectedLoanDueDate));
```

Comment:
Is this comparing the actual and expected loan dates without seconds or milliseconds?

Reply:
Yes it is.

Reply:
Why does the comparison need to ignore those parts of the due date?

Reply:
This is because the elapsed time between setting each date included a discrepancy of milliseconds, which was causing the test to fail. By flooring the milliseconds the tests (and we threw in the seconds for good measure) we were able to avoid this, while still ensuring the integrity of logic.

Reply:
The `TextDateTimeMatchers` class provides the `withinSecondsAfter` and `withinSecondsBefore` matchers which might be a more intent revealing alternative to this.